packages/template-retail-react-app/app/utils/product-utils.js

```diff
@@ -35,19 +35,31 @@ export const getDisplayVariationValues = (variationAttributes, values = {}) => {
 }
 
 /**
- * This function extract the promotional price from a product. If there are more than one price, the smallest price will be picked
+ * This function extract the list price and current price of a product
+ * If a product has promotional price, it will prioritize that value for current price
+ * List price will take the highest value among the price book prices
  * @param {object} product - product detail object
- * @returns {{discountPrice: number, basePrice: number | string}}
+ * @returns {{listPrice: number, currentPrice: number}}
  */
 export const getDisplayPrice = (product) => {
-    const basePrice = product?.pricePerUnit || product?.price
     const promotionalPriceList = product?.productPromotions
         ?.map((promo) => promo.promotionalPrice)
         .filter((i) => i !== null && i !== undefined)
-    // choose the smallest price among the promotionalPrice
-    const discountPrice = promotionalPriceList?.length ? Math.min(...promotionalPriceList) : null
+    const promotionalPrice = promotionalPriceList?.length ? Math.min(...promotionalPriceList) : null
+    // prioritize variant promotionalPrice over standard price
+    let currentPrice = promotionalPrice || product?.price
```

Comment:
How does tiered pricing work into this calculation. Form briefly looking at the API, tiered pricing is about having lower prices for when you are buying higher quantities. I don't think we are taking quantity in to account at all here. 

Imagine that on the PDP someone has increased their quality counter to like 100.. and at 100 items the price tier make the price lower than the promotional price. So in general we don't want to prioritize promotional prices, we simply want to choose the lowest price given that the product might have promo's, price ranges, tiers.

Reply:
Added logic to take tieredPrice into the account